### DISCUSSION TITLE
[Bug]: Permanent redirection for https does not work

### DISCUSSION THREAD

**Discussion by hamanuha** [coollabsio/coolify#5842] [Improvement Requests]
### Description

I‘m trying to get a 301 redirection instead of a 302 for the http to https redirection for a static service.
I added this to the Traefik labels:

`traefik.http.middlewares.redirect-to-https.redirectscheme.permanent=true`

But then the redirect isn’t working at all and I get a 404 instead if I access the site with http.

### Minimal Reproduction (if possible, example repository)

- Create a static service
- Enable http > https redirection
- Add the label provided above
- Redeploy
- Visit the http url

### Version

beta.323

### Cloud?

- [X] Yes
- [ ] No

**Comment by hamanuha** [comment]
I solved this. This has to be added to all services because that’s a global rule and then there is a mismatch.

Can this label be moved out of the service definition into the global config? Or should it be scoped to each service?

**Comment by stugreen13** [comment]
+1 same issue here. Another way to solve it is to give a different name to the middleware for the service you want to change, e.g. changing `redirect-to-https` to an arbitrary name like `myservicename-redirect-to-https` 


example:

```
traefik.http.middlewares.myservicename-redirect-to-https.redirectscheme.scheme=https
traefik.http.middlewares.myservicename-redirect-to-https.redirectscheme.permanent=true

# also update routers to use the new name
traefik.http.routers.http-0-bk0k8g80sgo4c8o0g0g0008c.middlewares=myservicename-redirect-to-https
traefik.http.routers.http-1-bk0k8g80sgo4c8o0g0g0008c.middlewares=myservicename-redirect-to-https

**Comment by hamanuha** [comment]
The redirect should probably be permanent by default.
Does a non default redirection from http to https even makes sense for anything?
